### PR TITLE
ADDED pdf417 support

### PR DESCRIPTION
**Pull request by yhauxell**
ADDED pdf417 support
ADDED latin characters support

### CODE REVIEW

File: commands.js [hunk 1]
```diff
@@ -213,6 +213,41 @@ _.GSV0_FORMAT = {
   GSV0_DWDH     : '\x1d\x76\x30\x03'
 };
 
+_.PDF417_FORMAT = {
```
Comment (uyu423) [inline commands.js]: Is the pdf417 format a format that works on printers with all esc/pos specifications?

File: examples/pdf417.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const printer = new escpos.Printer(device);
+
+let data = 'Lorem Ipsum es simplemente el texto de relleno de las imprentas y archivos de texto. Lorem Ipsum ha sido el texto de relleno estándar de las industrias desde el año 1500, cuando un impresor (N. del T. persona que se dedica a la imprenta) desconocido usó una galería de textos y los mezcló de tal manera que logró hacer un libro de textos especimen. No sólo sobrevivió 500 años, sino que tambien ingresó como texto de relleno en documentos electrónicos, quedando esencialmente igual al original. Fue popularizado en los 60s con la creación de las hojas "Letraset", las cuales contenian pasajes de Lorem Ipsum, y más recientemente con software de autoedición, como por ejemplo Aldus PageMaker, el cual incluye versiones de Lorem Ipsum.';
+
+
+device.open(function () {
+
+    printer
+        .align('rt')
+        .pdf417(data)
```
Comment (uyu423) [inline examples/pdf417.js]: I wonder if I need to create a new `Printer.pdf417()` method even though the `Printer.barcode()` method already exists in escpos.

File: package.json [hunk 3]
```diff
@@ -31,9 +31,9 @@
   },
   "optionalDependencies": {
     "get-pixels": "*",
-    "serialport": "*",
     "qr-image": "*",
-    "usb": "*"
+    "serialport": "*",
+    "usb": "git+https://github.com/yhauxell/node-usb.git"
```
Comment (uyu423) [inline package.json]: Why would you want to use the latest usb module without using a stable version of npm?

File: printer.js [hunk 4]
```diff
@@ -98,8 +99,8 @@ Printer.prototype.println = function(content){
  * @param  {[String]}  encoding [description]
  * @return printer instance
  */
-Printer.prototype.text = function(content, encoding){
-  return this.print(iconv.encode(content + _.EOL, encoding || 'GB18030'));
+Printer.prototype.text = function (content, encoding) {
+  return this.print(iconv.encode(content + _.EOL, encoding || 'iso-8859-1'));
```
Comment (uyu423) [inline printer.js]: Changing the default encoding character setting is still being discussed. #64 #67 #69

File: printer.js [hunk 5]
```diff
@@ -335,6 +336,60 @@ Printer.prototype.qrcode = function(code, version, level, size){
...
+  this.buffer.write(_.PDF417_FORMAT.DATA);
+  this.write(code);
+  //print
+  this.buffer.write(_.GS + _.PDF417_FORMAT.START);
+  this.buffer.write(_.PDF417_FORMAT.PRINT);
+
+  return this;
+}
+
+Printer.prototype.codetable = function (code) {
```
Comment (uyu423) [inline printer.js]: What is the purpose of the codetable?

**Comment by yhauxell** [comment]
Coming to your questions: 
- Yes, PDF417 is a 2D barcode, its commands are available for all ESC/POS printers
- Yes, I believe we can extend barcode method instead of create a new one. 
- I started using node-usb instead of usb because of some windows.h file issue due gyp build with electron. better to re-check.
- codetable function enables to write spanish characters as ñ, á, etc.

I´ll work more in this solution before create a new pull request, sorry for taking your time and thanks for your comments.

**Pull request closed without merging**